assume/reinforcement_learning/network_architecture.py

```diff
@@ -0,0 +1,188 @@
+            self.FC2_2 = nn.Linear(512, 256, dtype=float_type)
+            self.FC2_3 = nn.Linear(256, 128, dtype=float_type)
+            self.FC2_4 = nn.Linear(128, 1, dtype=float_type)
+        else:
+            self.FC2_1 = nn.Linear(self.obs_dim + self.act_dim, 1024, dtype=float_type)
+            self.FC2_2 = nn.Linear(1024, 512, dtype=float_type)
+            self.FC2_3 = nn.Linear(512, 128, dtype=float_type)
+            self.FC2_4 = nn.Linear(128, 1, dtype=float_type)
+
+    def forward(self, obs, actions):
+        """
+        Forward pass through the network, from observation to actions.
+        """
+        xu = th.cat([obs, actions], 1)
+
+        x1 = F.relu(self.FC1_1(xu))
+        x1 = F.relu(self.FC1_2(x1))
+        x1 = F.relu(self.FC1_3(x1))
+        x1 = self.FC1_4(x1)
+
+        x2 = F.relu(self.FC2_1(xu))
+        x2 = F.relu(self.FC2_2(x2))
+        x2 = F.relu(self.FC2_3(x2))
+        x2 = self.FC2_4(x2)
+
+        return x1, x2
+
+    def q1_forward(self, obs, actions):
+        """
+        Only predict the Q-value using the first network.
+        This allows to reduce computation when all the estimates are not needed
+        (e.g. when updating the policy in TD3).
+
+        Args:
+            obs (torch.Tensor): The observations
+            actions (torch.Tensor): The actions
+
+        """
+        x = th.cat([obs, actions], 1)
+        x = F.relu(self.FC1_1(x))
+        x = F.relu(self.FC1_2(x))
+        x = F.relu(self.FC1_3(x))
+        x = self.FC1_4(x)
+
+        return x
+
+
+class Actor(nn.Module):
+    """
+    Parent class for actor networks.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+
+class MLPActor(Actor):
+    """
+    The neurnal network for the MLP actor.
+    """
+
+    def __init__(self, obs_dim: int, act_dim: int, float_type):
+        super().__init__()
+
+        self.FC1 = nn.Linear(obs_dim, 256, dtype=float_type)
+        self.FC2 = nn.Linear(256, 128, dtype=float_type)
+        self.FC3 = nn.Linear(128, act_dim, dtype=float_type)
+
+    def forward(self, obs):
+        x = F.relu(self.FC1(obs))
+        x = F.relu(self.FC2(x))
+        x = F.softsign(self.FC3(x))
+        # x = th.tanh(self.FC3(x))
+
+        return x
+
+
+class LSTMActor(nn.Module):
+    """
+    The LSTM recurrent neurnal network for the actor.
+
+    Based on: "Multi-Period and Multi-Spatial Equilibrium Analysis in Imperfect Electricity Markets"
+    by Ye at al. (2019)
+    """
+
+    def __init__(self, obs_dim: int, act_dim: int, float_type):
+        super().__init__()
+        self.float_type = float_type
+        self.no_of_timeseries = 2  # TODO: variable no. of input timeseries
+        self.forecast_horizon = int(
+            (obs_dim - 2) / self.no_of_timeseries
+        )  # TODO: variable number of additional input features, currently 2 (marginal costs, capacity)
+
+        self.LSTM1 = nn.LSTMCell(self.no_of_timeseries, 8, dtype=float_type)
```

Comment:
not for now, but maybe some food for thoughts, we could just try this out with a GRU units instead of LSTM, which for our use case would probably be sufficient and much faster.

Reply:
yes, we thought about GRU (or xLSTM) as well. right now we're not focussing on implementing another variation of the actor network.